src/main/java/com/epam/ms/service/DefaultNutritionProgramService.java

```diff
@@ -55,4 +74,10 @@ private void copyProgramData(DefaultNutritionProgram from, DefaultNutritionProgr
         to.setCalories(from.getCalories());
         to.setNumberOfDays(from.getNumberOfDays());
     }
+
+    private void sendEventToQueue(String eventKey, DefaultNutritionProgram program) throws JsonProcessingException {
```

Comment:
usually on production we don't really throw Exception through all the layers
we wrap it in some RuntimeException (ServiceException, for instance)
```
try {
   template.convertAndSend(NUTRITION_EVENTS_QUEUE, jacksonMapper.writeValueAsString(event));
} catch (JsonProcessingException e) {
   throw new ServiceException(String.format("event %s could not be serialized", event), e);
}
```

Reply:
Thank you, done